tsdb/head_append.go

```diff
@@ -1295,94 +1302,103 @@ func (a *headAppender) Commit() (err error) {
 			a.head.metrics.chunksCreated.Inc()
 		}
 
-		series.cleanupAppendIDsBelow(a.cleanupAppendIDsBelow)
-		series.pendingCommit = false
-		series.Unlock()
+		acc.series.cleanupAppendIDsBelow(a.cleanupAppendIDsBelow)
+		acc.series.pendingCommit = false
+		acc.series.Unlock()
 	}
+}
 
+// commitFloatHistograms processes and commits float histograms to the headAppender.
+// It iterates over the float histograms in the headAppender and attempts to append
+// each histogram to its corresponding series. Depending on the result of the append
+// operation, it updates various metrics and counters in the appenderCommitContext.
+//
+// The function handles different error scenarios such as out-of-order samples,
+// out-of-bounds samples, and too-old samples. It also manages the insertion of
+// out-of-order samples and updates the write-ahead log (WAL) records accordingly.
+//
+// Metrics and counters updated:
```

Comment:
This bit is different from the other two commit* functions, please sync the comments. Maybe we could just have the long comment at commitSamples and mention there that commitHistograms and commitFloatHistograms do the same. And from these two refer back to commitSamples.

Reply:
@krajorama thanks for you feedback, yes, this makes sense.
Can you check and see wdyt?

Reply:
looks good